### ISSUE TITLE
Memory leaked with Skybox and IndirectLight on Android

### ISSUE THREAD

**Issue by August1996** [google/filament#6412]
<img width="1109" alt="image" src="https://user-images.githubusercontent.com/15972489/209832224-1ea295d3-9432-4c7f-8dc6-4b5a57077d3a.png">

As the picture showing, Creating `Skybox` or `IndirectLight` by `KTXLoader.createXXX`  will create `cubemap` Texture.
Normally, we may use `Engine.destroy` to release the `Skybox` or `IndirectLight`, but the `cubemap` Texture will not be destroyed unless the engine is destroyed or destroying the `cubemap` Texture explicitly(it is very oddly because nobody can realize it).

**Comment by romainguy** [comment]
Note: this only happens with the utility functions.

**Comment by pixelflinger** [comment]
Thanks for the report. This is a bad one, basically the API is broken.

**Comment by romainguy** [comment]
We can fix it easily in the Java case. Instead of a single long we should return two (the two pointers) and have the Java layer perform the additional destroy as needed.

**Comment by pixelflinger** [comment]
@romainguy I don't think it will work, because the java (kotlin actually) layer doesn't store the long, it just returns it to the caller:

```
    fun createIndirectLight(engine: Engine, buffer: Buffer, options: Options = Options()): IndirectLight {
        val nativeEngine = engine.nativeObject
        val nativeIndirectLight = nCreateIndirectLight(nativeEngine, buffer, buffer.remaining(), options.srgb)
        return IndirectLight(nativeIndirectLight)
    }
```

**Comment by romainguy** [comment]
We can store that in the Java IndirectLight object. It's very doable.

**Comment by pixelflinger** [comment]
Okey but we now made all the Java objects bigger just to support that broken API -- And we're breaking the model of java objects being "dumb", now these have logic.
